runtime/vm/jvminit.c

```diff
@@ -2376,6 +2386,10 @@ static UDATA checkArgsConsumed(J9PortLibrary* portLibrary, J9VMInitArgs* j9vm_ar
 			} else if (HAS_MAPPING( j9vm_args, i) && (MAPPING_FLAGS( j9vm_args, i ) & INVALID_OPTION)) {
 				j9nls_printf(PORTLIB, J9NLS_ERROR, J9NLS_VM_INVALID_CMD_LINE_OPT, optString);
 			} else {
+				/* If ignoreUnrecognizedXXOption is set to JNI_TRUE, we should ignore any options that start with -XX */
+				if (ignoreUnrecongizedXXOption && (0 == strncmp(optString, "-XX:", 4))) {
```

Comment:
comment and code don't match - comment says `-XX`, code says `-XX:`.  Which is the intended behaviour?

Reply:
-XX:, I'll change the comment

Reply:
I expected the answer would go the other way and the option would ignore all -XX options.  Why require the `:`?

Reply:
If I understand correctly, the original issue/enhancement was to report/ignore any issues that start with `-XX:` and not `-XX`

Reply:
That's an oversight.  Most `-XX` options are of the form `-XX:[+-]Something` so I think the original issue is overly specific.  Let's actually match on `-XX` to ensure the behaviour matches the option name

Reply:
Will do. I'll change the PR accordingly.

Reply:
This would be a change in behavior. The VM has only ever ignored options that start with `-XX:`

https://github.com/eclipse/openj9/blob/278be6c0ff617069c6fa2038ffac7afbf4cb5989/runtime/vm/jvminit.c#L183

https://github.com/eclipse/openj9/blob/278be6c0ff617069c6fa2038ffac7afbf4cb5989/runtime/oti/jvminit.h#L292

Reply:
A change in the default behavior.

Reply:
I'm OK with changing the corner case here so the behaviour is more consistent.

A quick test with the RI shows that `java -XXFoo -XX:+IgnoreUnrecognizedVMOptions -version` starts just fine.  We should have have the same behaviour when ignoring options.

Reply:
And we do. `java -XXFoo -XX:+IgnoreUnrecognizedVMOptions -version` starts with with OpenJ9 as well. That isn't what this PR is about.